swarm/src/connection/pool.rs

```diff
@@ -297,7 +297,7 @@ pub enum PoolEvent<THandler: IntoConnectionHandler> {
         id: ConnectionId,
         peer_id: PeerId,
         /// The produced event.
-        event: THandlerOutEvent<THandler>,
+        event: <<THandler as IntoConnectionHandler>::Handler as ConnectionHandler>::OutEvent,
```

Comment:
I temporarily inlined this one so that I can delete the 2nd type alias named `THandlerOutEvent` defined in `swarm::behaviour`.

This will be simplified to `ConnectionHandler::OutEvent` in #3254.